doc/mgmt/SONiC_overlay_ConfigDB.md

```diff
@@ -0,0 +1,234 @@
+  + [6.1 Phase 1](#6-1-phase-1)
+  + [6.2 Phase 1](#6-2-Phase-2)
+- [7 References](#7-references)
+
+# About this Manual
+
+This document provides a detailed description on the new features for:
+
+- Overlay Config DB.
+- Yang model default value overlay.
+- Static config overlay.
+- swss-common API change.
+
+## SONiC issue solved by this feature
+
+- Still using old default value and config from j2 template after OS upgrade:
+  - Following config may update after OS upgrade:
+    - Default value: defined in yang model.
+    - Static config: defined in j2 template.
+  - Currently all config stored in config DB, so config above can't update after OS upgrade.
+- Potential risk, Yang model default value conflict with hardcoded value:
+  - Default value hardcoded in source code.
+  - Yang model default value not used.
+- SONiC utilities not support get user config and full config:
+  - Vender OS have different show command:
+    - show running: only return user config.
+    - show running all: user config, default value from yang model, and config from j2 template.
+  - Currently SONiC only support 'show running'
+
+# 1 Functional Requirement
+
+## 1.1 swss-common return default value from Yang model overlay
+
+- Return default value is optional.
+  - Application can read config without default value, also can read config with default value.
+- Backward compatibility with existed code and applications.
+
+## 1.2 swss-common return static config from static config overlay
+
+- Static buffer config stored in static config tables.
+- Return static config is optional.
+  - Application can decide read config with static config or not.
+- Backward compatibility with existed code and applications.
+  - For backward compatibility when initialize buffer config from graph, config will be write to both config tables and static config tables.
+  - After code migrate to use static buffer config, static config will only write to static config tables.
+  - Static config support delete/revert operation:
+    - In some user scenario, user need delete a static config, and also may add config back later.
+    - For delete operation, data in static config table will not be delete.
+    - Static overlay use STATIC_CONFIG_DELETE table to handle dlete/revert:
+      - When delete a static config item, static overlay will add the item key to STATIC_CONFIG_DELETE table.
+      - When read static config, all item in STATIC_CONFIG_DELETE will not include in result.
+
+# 2 Design
+
+- Layered config DB design diagram:
+
+<img src="./images/swss-common-layer.png"  />
+
+- API design diagram:
+
+<img src="./images/swss-common-default-value.png"  />
+
+## 2.1 Considerations
+
+### How to get default value
+
+|                                                               | Pros                                              | Cons                                                                                                                                              |
```

Comment:
>| [](http://example.com/codeflow?start=0&length=1)

It's good to provide table. However not clear which option do you choose

Reply:
Fixed, move the current design part to above this section.